lib/auth/helpers.go

```diff
@@ -102,14 +102,16 @@ func (cfg *TestAuthServerConfig) CheckAndSetDefaults() error {
 	return nil
 }
 
-// CreateUploaderDir creates directory for file uploader service
-func CreateUploaderDir(dir string) error {
-	if err := os.MkdirAll(filepath.Join(dir, teleport.LogsDir, teleport.ComponentUpload,
-		events.StreamingLogsDir, apidefaults.Namespace), teleport.SharedDirMode); err != nil {
-		return trace.ConvertSystemError(err)
+// CreateUploaderDirs creates the directory structure for the file uploader service.
+func CreateUploaderDirs(dir string) error {
+	var errs []error
+
+	for _, dirname := range []string{events.StreamingSessionsDir, events.CorruptedSessionsDir} {
+		path := filepath.Join(dir, teleport.LogsDir, teleport.ComponentUpload, dirname, apidefaults.Namespace)
+		errs = append(errs, os.MkdirAll(path, teleport.SharedDirMode))
```

Comment:
Yeah because on first glance `ConvertSystemError` didn't seem to handle nil, and this is only a test helper so I thought I could get away with it. I'll take a closer look.